packages/backend-function/src/runtime/get_amplify_clients_configuration.ts

```diff
@@ -125,6 +128,38 @@ export type DataClientReturn<T> = T extends DataClientEnv
   ? DataClientConfig
   : DataClientError;
 
+const extendEnv = (
+  env: DataClientEnv & Record<string, unknown>,
+  dataName: string
+): ExtendedAmplifyClientEnv => {
+  const bucketName = `${dataName}${dataBucketNameContent}`;
+  const keyName = `${dataName}${dataKeyNameContent}`;
+  const endpointName = `${dataName}${dataEndpointNameContent}`;
+  if (
+    !(
+      bucketName in env &&
+      keyName in env &&
+      endpointName in env &&
+      typeof env[bucketName] === 'string' &&
+      typeof env[keyName] === 'string' &&
+      typeof env[endpointName] === 'string'
+    )
+  ) {
+    throw new Error('The data environment variables are malformed');
```

Comment:
Can you also add serialized `env` and `dataName` in the error message.

Reply:
Added